tests/integration/pilot/dnsautoallocation/main_test.go

```diff
@@ -0,0 +1,56 @@
+//go:build integ
+// +build integ
+
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package dnsautoallocation
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo/common/deployment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var (
+	i istio.Instance
+
+	// Below are various preconfigured echo deployments. Whenever possible, tests should utilize these
+	// to avoid excessive creation/tear down of deployments. In general, a test should only deploy echo if
+	// its doing something unique to that specific test.
+	apps = deployment.SingleNamespaceView{}
+)
+
+func TestMain(m *testing.M) {
```

Comment:
I appreciate the effort put into a test here. However, I think there may be some high cost vs value for this that may be better suited for a different type of test.

Every TestMain adds 1-2mins to each CI job. Which isn't that long, but its a death by 1000 cuts situation; 2 minutes for a single test doesn't provide a strong cost:value tradeoff in most cases.

It does seem like most of this could be handled by unit tests. The fix here is that "with autoallocation, two SEs without `address` should behave like two SE with `address`". We do already have integration tests for SE (although they could probably be improved -- I certainly wouldn't mind additional test cases their to existing suites, if you are interested), so we could just test the two are equivilent which could be done in unit tests.

In particular, tests in https://github.com/istio/istio/blob/0fb12bfcfe582458b70b5cb911fb0307ee927d3c/pilot/pkg/networking/core/v1alpha3/sidecar_simulation_test.go#L76 follow a pattern that provides what looks *a lot* like an integration tests, but is done through static xds analysis rather than full deployments; this may provide much of the same coverage and run ~instantly